src/test/java/jcifs/tests/SymLinkTest.java

```diff
@@ -0,0 +1,188 @@
+import jcifs.smb.SmbFile;
+
+
+/**
+ * The actual SMB share must be configured in the system properties
+ * with the 'real' server settings for testing.
+ *
+ * You must create the following folder structures in your SMB share
+ * for a successful completion of these unit tests:
+ *
+ * <pre class="code">
+ *  smbSource/
+ *  |-- subSmbSource/
+ *      |-- subSmbSource1.txt - contains 'subSource1'
+ *      |-- subSmbSource2.txt - contains 'subSource2'
+ *  smbTarget/
+ *  |-- foo.file - contains 'foofoofoo'
+ *  symlinkTestDir/ - symbolic link to directory smbSource/subSmbSource/
+ *                    using 'mklink /D symlinkTestDir <share root>\smbSource\subSmbSource\'
+ *  symlinkTestFoo.file - symbolic link to file smbTarget/foo.file
+ *                        using 'mklink symlinkTestFoo.file <share root>\smbTarget\foo.file'
+ * </pre>
+ *
+ * @author Gregory Bragg
+ */
+@RunWith ( Parameterized.class )
+@SuppressWarnings ( "javadoc" )
+public class SymLinkTest extends BaseCIFSTest {
+
+    private static final Logger log = LoggerFactory.getLogger(SymLinkTest.class);
+
+    private static final String SMB_FILE_SEPARATOR = "/";
+
+    public SymLinkTest ( String name, Map<String, String> properties ) {
+        super(name, properties);
+    }
+
+
+    @Parameters ( name = "{0}" )
+    public static Collection<Object> configs () {
+        return getConfigs("smb1", "noUnicode", "forceUnicode", "noNTStatus", "noNTSmbs", "smb2", "smb30", "smb31");
+    }
+
+
+    @Test
+    public void testSymlink1() throws MalformedURLException, CIFSException {
+        SmbFile smbFile = createSession();
+        assertNotNull(smbFile);
+
+        SmbFile[] files = smbFile.listFiles();
+        log.info(Arrays.toString(files));
+        assertNotNull(files);
+        assertTrue("No share found", files.length > 0);
+
+        for (SmbFile file : files) {
+            assertNotNull(file);
+
+            if (file.isDirectory()) {
+                log.info("Resource [" + file.getName() + "] is a directory.");
+            }
+            else {
+                log.info("Resource [" + file.getName() + "] is not a directory.");
+            }
+
+            if (file.isFile()) {
+                log.info("Resource [" + file.getName() + "] is a file.");
+            }
+            else {
+                log.info("Resource [" + file.getName() + "] is not a file.");
+            }
+
+            if (file.isSymlink()) {
+                log.info("Resource [" + file.getName() + "] is a symbolic link.");
+            }
+            else {
+                log.info("Resource [" + file.getName() + "] is not a symbolic link.");
+            }
+
+            try {
+                InputStream is = file.getInputStream();
+                assertNotNull(is);
+
+                if (file.isDirectory()) {
+                    log.info("file.list() -> {}", Arrays.toString(file.list()));
+
+                    if (file.getName().equals("symlinkTestDir/")) {
```

Comment:
These names could be specified in a configuration property, then the test also could be easily skipped when the setup was not done/symlinks are not supported.

Reply:
Same comment as below...